lib/packages/docutils/rst.nim

```diff
@@ -1246,15 +1246,44 @@ proc isUrl(p: RstParser, i: int): bool =
     p.tok[i+3].kind == tkWord and
     p.tok[i].symbol in ["http", "https", "ftp", "telnet", "file"]
 
+proc checkParen(token: Token, parensStack: var seq[char]): bool {.inline.} =
```

Comment:
completely optional: can you split this into a more general reusable proc and then call it, and then in future work we can move this reusable proc to strbasics or strutils? (similar to https://dlang.org/library/std/algorithm/searching/balanced_parens.html, and same functionality as what you have)
```nim
proc balancedParens(a: openArray[char], parensStack: var seq[char]): bool =
  # xxx in future work, move to stdlib
proc checkParen(token: Token, parensStack: var seq[char]): bool {.inline.} =
  call balancedParens somehow
```

Reply:
The idea of `balancedParens` is very different: it check whether parens are balanced or not.

`checkParen` does not always forbid unbalanced parens: e.g. `([)` and `(])` are legal, and `checkParen` will return 3 values `False,False,True` for those 3 tokens.